qt/scientific_interfaces/Inelastic/QENSFitting/FitOutput.cpp

```diff
@@ -118,25 +107,20 @@ void FitOutput::clear() {
 
 void FitOutput::addOutput(const Mantid::API::WorkspaceGroup_sptr &resultGroup,
                           Mantid::API::ITableWorkspace_sptr parameterTable,
-                          const Mantid::API::WorkspaceGroup_sptr &resultWorkspace) {
-  m_parameters = extractParametersFromTable(parameterTable);
+                          const Mantid::API::WorkspaceGroup_sptr &resultWorkspace, FitDomainIndex fitDomainIndex) {
+  TableRowExtractor extractRowFromTable(std::move(parameterTable));
   m_resultGroup = resultGroup;
   m_resultWorkspace = resultWorkspace;
-  m_outputResultLocations.clear();
-  for (size_t index = 0; index < resultGroup->size(); index++) {
-    m_outputResultLocations.emplace(index, ResultLocationNew(resultGroup, WorkspaceID{static_cast<size_t>(index)}));
+  if (resultGroup->size() == 1u) {
```

Comment:
The for loop does an update of all spectra results by just looping through the size of the resultGroup. However, in the case of doing a single fit, looping through the results group (which would have a size of 1) would update only the result at index 0. This will not always be the correct index to update, because a single fit can be done on any of the spectra - so in this case we need to use the specific FitDomainIndex passed into the function